test/unit/app/controllers/permissions/permissions-controller-test.js

```diff
@@ -0,0 +1,1392 @@
+        { origin: ORIGINS.a, resolve: noop, reject: noop },
+        'pending approvals should have expected entry'
+      )
+      assert.deepEqual(
+        permController.pendingApprovals.get(id3),
+        { origin: ORIGINS.c, resolve: noop, reject: noop },
+        'pending approvals should have expected entry'
+      )
+
+      assert.ok(
+        permController.pendingApprovalOrigins.has(ORIGINS.a),
+        'pending approvals origins has expected item',
+      )
+      assert.ok(
+        permController.pendingApprovalOrigins.has(ORIGINS.c),
+        'pending approvals origins has expected item',
+      )
+    })
+
+    it('should throw if adding origin twice', function () {
+
+      const id = nanoid()
+      const origin = ORIGINS.a
+
+      permController._addPendingApproval(id, origin, noop, noop)
+
+      const otherId = nanoid()
+
+      assert.throws(
+        () => permController._addPendingApproval(otherId, origin, noop, noop),
+        ERRORS.pendingApprovals.duplicateOriginOrId(otherId, origin)
+      )
+
+      assert.equal(
+        permController.pendingApprovals.size, 1,
+        'pending approvals has single entry',
+      )
+
+      assert.equal(
+        permController.pendingApprovalOrigins.size, 1,
+        'pending approvals origins has single item',
+      )
+
+      assert.deepEqual(
+        permController.pendingApprovals.get(id),
+        { origin, resolve: noop, reject: noop },
+        'pending approvals should have expected entry'
+      )
+
+      assert.ok(
+        permController.pendingApprovalOrigins.has(origin),
+        'pending approvals origins has expected item',
+      )
+    })
+
+    it('should throw if adding id twice', function () {
+
+      const id = nanoid()
+      const origin = ORIGINS.a
+
+      permController._addPendingApproval(id, origin, noop, noop)
+
+      assert.throws(
+        () => permController._addPendingApproval(id, ORIGINS.b, noop, noop),
+        ERRORS.pendingApprovals.duplicateOriginOrId(id, ORIGINS.b)
+      )
+
+      assert.equal(
+        permController.pendingApprovals.size, 1,
+        'pending approvals has single entry',
+      )
+
+      assert.equal(
+        permController.pendingApprovalOrigins.size, 1,
+        'pending approvals origins has single item',
+      )
+
+      assert.deepEqual(
+        permController.pendingApprovals.get(id),
+        { origin, resolve: noop, reject: noop },
+        'pending approvals should have expected entry'
+      )
+
+      assert.ok(
+        permController.pendingApprovalOrigins.has(origin),
+        'pending approvals origins has expected item',
+      )
+    })
+  })
+
+  describe('miscellanea and edge cases', function () {
+
+    let permController, notifications
+
+    beforeEach(function () {
+      notifications = initNotifications()
+      permController = initPermController(notifications)
+    })
+
+    it('notifyDomain handles notifications other than accountsChanged', async function () {
```

Comment:
`notifyDomain`, as a public method, probably deserves its own `describe` block

Reply:
Fixed in aec89f9